python/cuml/cuml/ensemble/randomforestclassifier.py

```diff
@@ -237,18 +237,13 @@ def fit(self, X, y, *, convert_dtype=True) -> "RandomForestClassifier":
             check_dtype=[np.float32, np.float64],
             order="F",
         ).array
-        y_m = input_to_cuml_array(
-            y,
-            convert_to_dtype=(np.int32 if convert_dtype else None),
-            check_dtype=np.int32,
-            check_rows=X_m.shape[0],
-            check_cols=1,
-        ).array
+        y = input_to_cupy_array(y, check_rows=X_m.shape[0], check_cols=1).array
+        check_classification_targets(y)
 
-        self.classes_ = cp.unique(y_m)
+        classes, y = cp.unique(y, return_inverse=True)
```

Comment:
Does `return_inverse=True` replaces `make_monotonic`? Would be great to comment this, so that this is not inadvertently removed.